Yoast/yoast-components
# Stories/267 feed reader

## Summary

This PR can be summarized in the following changelog entry:

* Adds a `getFeed` util to allow fetching and parsing RSS feeds.

## Relevant technical choices:

* Uses https://github.com/bobby-brennan/rss-parser which seems well supported including in the browser. https://github.com/danmactough/node-feedparser is an alternative but does not seem to explicitly support the browser.

## Test instructions

This PR can be tested by following these steps:

* Have a Wordpress site that has CORS headers on it's RSS feed. ( You can add this to your own by adding `header("Access-Control-Allow-Origin: *");` on line 9 in `wp-includes/feed-rss2.php` ).
* Call `getFeed` with the feed URL and the maximum number of entries you wish returned. It returns a `Promise` object.
* The `Promise` should resolve with a `Feed` object as documented.

Fixes #267.
